packages/@react-spectrum/listview/src/ListView.tsx

```diff
@@ -0,0 +1,165 @@
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {AriaLabelingProps, CollectionBase, DOMProps, DOMRef, StyleProps} from '@react-types/shared';
+import {classNames, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+import {GridCollection, useGridState} from '@react-stately/grid';
+import {GridKeyboardDelegate, useGrid} from '@react-aria/grid';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {ListState, useListState} from '@react-stately/list';
+import listStyles from './listview.css';
+import {ListViewItem} from './ListViewItem';
+import {ListViewLayout} from '@react-stately/layout';
+import {ProgressCircle} from '@react-spectrum/progress';
+import React, {ReactElement, useContext, useMemo} from 'react';
+import {useCollator, useLocale, useMessageFormatter} from '@react-aria/i18n';
+import {useProvider} from '@react-spectrum/provider';
+import {Virtualizer} from '@react-aria/virtualizer';
+
+
+export const ListViewContext = React.createContext(null);
+
+export function useListLayout<T>(state: ListState<T>) {
+  let {scale} = useProvider();
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let layout = useMemo(() =>
+      new ListViewLayout<T>({
+        estimatedRowHeight: scale === 'large' ? 48 : 32,
+        padding: 0,
+        collator
+      })
+    , [collator, scale]);
+
+  layout.collection = state.collection;
+  layout.disabledKeys = state.disabledKeys;
+  return layout;
+}
+
+interface ListViewProps<T> extends CollectionBase<T>, DOMProps, AriaLabelingProps, StyleProps {
+  isLoading?: boolean,
+  renderEmptyState?: () => JSX.Element,
+  transitionDuration?: number
+}
+
+function ListView<T extends object>(props: ListViewProps<T>, ref: DOMRef<HTMLDivElement>) {
+  let {
+    transitionDuration = 0
+  } = props;
+  let domRef = useDOMRef(ref);
+  let {collection} = useListState(props);
+  let formatMessage = useMessageFormatter(intlMessages);
+
+  let {styleProps} = useStyleProps(props);
+  let {direction} = useLocale();
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let gridCollection = useMemo(() => new GridCollection({
+    columnCount: 1,
+    items: [...collection].map(item => ({
+      type: 'item',
+      childNodes: [{
+        ...item,
+        index: 0,
+        type: 'cell'
+      }]
+    }))
+  }), [collection]);
+  let state = useGridState({
+    ...props,
+    collection: gridCollection
+  });
+  let layout = useListLayout(state);
+  let keyboardDelegate = new GridKeyboardDelegate({
```

Comment:
Probably should use `useMemo` here. This is creating a new delegate on each render.